AppInspector.CLI/CLICmdOptions.cs

```diff
@@ -19,6 +19,37 @@ public class CLICommandOptions : CommandOptions
         public string OutputFileFormat { get; set; } = "text";
     }
 
+    /// <summary>
+    /// CLI command distinct arguments
+    /// </summary>
+    [Verb("get-tags", HelpText = "Inspect source directory/file/compressed file (.tgz|zip) against defined characteristics for unique tags")]
+    public class CLIGetTagsCommandOptions : CLICommandOptions
+    {
+        [Option('s', "source-path", Required = true, HelpText = "Source file or directory to inspect")]
+        public string? SourcePath { get; set; }
+
+        [Option('r', "custom-rules-path", Required = false, HelpText = "Custom rules file or directory path")]
+        public string? CustomRulesPath { get; set; }
+
+        [Option('i', "ignore-default-rules", Required = false, HelpText = "Exclude default rules bundled with application", Default = false)]
+        public bool IgnoreDefaultRules { get; set; }
+
+        [Option('c', "confidence-filters", Required = false, HelpText = "Output only matches with specified confidence <value>,<value> [high|medium|low]", Default = "high,medium")]
+        public string ConfidenceFilters { get; set; } = "high,medium";
+
+        [Option('k', "file-path-exclusions", Required = false, HelpText = "Exclude source files (none|default: sample,example,test,docs,lib,.vs,.git)", Default = "sample,example,test,docs,lib,.vs,.git")]
```

Comment:
We should probably also exclude binary files -- e.g. .exe, .dll, .v2, .pdb, unless there's a nice fast way of maybe sampling the distribution of high ascii chars from the first 256 bytes of the file, or something like that?

Because I ran this against a basic VS source project and since the pdb, exe, etc. got pulled in, the resultant .html was huge.

Reply:
The old behavior is to ignore everything that didn't have a known extension. Is that a better approach?

Reply:
Hmm, we had to do TreatEverythingAsCode in order to get it to scan everything. Not sure what the best approach is, but with this PR, if you scan a directory with a .pdb in it, you'll get gobbly-gook in the html assuming some text matches a rule.

Reply:
I could only scan files that are valid UTF8?

Reply:
I added a check to look at the first 100 characters and if more than 50% of them are control characters it will skip the file.

Reply:
Checking a handful of binaries shows they're over 50% of the first 1024 characters, in particular a lot of \0 and this check works.  Open to any thoughts on tweaking how many characters to check and how many should be control as well.  50% may actually be a very high proportion. There shouldn't be many if any non-printable characters in source code.  Haven't checked the performance implications but could also check every character in the file (the streams should be in memory most of the time), this likely isn't that much extra processing to scan the file one more time if you consider how many regexes we run on the entire file.

Reply:
Rewrote this to match Perl's behavior which is "Is binary if it has any NULs or if its more than 30% non -printable characters"